tests/lib/drivers/test_aws_lambda.py

```diff
@@ -51,8 +61,12 @@ def test_deploy_w_provisions(
 
         image = aws_ecr_driver.get_image("0.0.1")
         aws_lambda_driver.deploy(image, Architecture.system())
+        self.create_log_group(
+            aws_lambda_driver.log_group
+        )  # Is this something that moto should be doing?
```

Comment:
https://docs.getmoto.org/en/latest/docs/services/cloudwatch.html

If it isn't exploding there's a good chance moto is handling this. The clients singleton all but ensures this. During test, everything is fed the test config and cannot escape moto.

Reply:
Unrelated follow up:

In my readings this way of doing the clients is an anti-pattern. It's sort of a poor-mans dependency injection.

The thing is, in my explorations of "proper" solutions, I came to feel that the medicine was worse than the disease.

Am I right? :man_shrugging: